cpp/loader/ndarray_cache_metadata.h

```diff
@@ -0,0 +1,94 @@
+#ifndef MLC_LLM_CPP_LOADER_NDARRAY_CACHE_METADATA_H_
```

Comment:
Remove this file, and instead in TVM Unity, move `ndarray_cache_metadata.h` to public API